enhancements/microshift/microshift-telemetry.md

```diff
@@ -0,0 +1,338 @@
+
+#### `cluster:usage:resources:sum`
+This metric reports the number of Kubernetes objects by resource type. Already
+supported in OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `resource`: name of the kubernetes resource. Values can be `pods`,
+  `namespaces`, `services`, `ingresses.networking.k8s.io`,
+  `routes.route.openshift.io`,
+  `customresourcedefinitions.apiextensions.k8s.io`.
+
+#### `cluster:usage:containers:sum`
+This metric reports the number of running containers. Already supported in
+OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+
+#### `microshift_version`
+This metric reports information about MicroShift version. New metric not
+supported in OpenShift.
+
+Labels:
+* `_id`: cluster ID. UUID string.
+* `version`: MicroShift version. Possible values are `\d{1}.\d{1,2}.\d{1,2}`.
+* `deployment_type`: Describes deployment type. Possible values are `rpm`,
+  `ostree`, `bootc`.
+* `deployment_id`: In case `deployment_type` is `ostree`, this will hold the
+  ostree commit. In case of `bootc`, include image SHA. Format is a 64
+  character SHA.
+
+
+Using the names and labels listed above, each sample requiring a float64
+and an int64 value, all metrics combined are under 2KB.
+
+In order to keep metrics synchronized with those of OpenShift, in terms of API
+support, MicroShift will be using CI:
+* A job shall deploy MicroShift with Telemetry enabled.
+* A job will use the staging endpoint for Telemetry API. No added traffic
+  and no confusing data when querying.
+* At the end of the job a new check shall be performed: check there is data
+  belonging to that specific cluster in the staging environment.
+
+### Sending metrics
+OpenShift is sending metrics through this API every 4.5 minutes. This is done
+to overcome staleness in Prometheus server on the Telemetry backend. Stale data
+may yield gaps when retrieving it, making querying and dashboards more nuanced
+and harder to read.
+
+MicroShift, by its nature, needs to cover different use cases and potential
+limitations: resource constrained environments where computing power is scarce
+or limited, unreliable networks where outages are common, slow networks where
+bandwidth is not the best, private networks where all external traffic is
+audited and maybe limited, applications that are stable once they start,
+disconnected clusters, etc. The dynamic behavior that metrics help uncover is
+only one of the many use cases MicroShift has.
+For these reasons, and because the intention of this enhancement is to get the
+basic structure of a metrics reporting system, MicroShift will start using 1h
+intervals to send metrics, which is a good compromise between resource usage
+and data granularity.
+
+When MicroShift starts it should send metrics once and then schedule them every
+hour.
+It will also send metrics when issued to stop, as part of the finish process.
+This helps in sending a last report and also hint when the cluster has been
+stopped.
```

Comment:
(nit) from experience: when a cluster stops reporting metrics, it's almost impossible to differentiate between the cluster has been shutdown and there's an issue in the telemetry exchange.

Reply:
Relaxed the wording to remove the hint about cluster being stopped.